server/dataset/cxg_dataset.py

```diff
@@ -403,7 +409,14 @@ def annotation_to_fbs_matrix(self, axis, fields=None):
                 else:
                     data = A.query(attrs=fields)[:]
 
+                categorical_dtypes = []
```

Comment:
same question as above - why is this not being done in the FBS encoder?  It is directly related to the FBS optimizations, not to the data model or CXG.

You are using the schema as a hint for what might have originally been a Pandas Categorical, but that hint could be passed from this layer to the underlying FBS encoder.

Perhaps even better the FBS encoder could decide on its own whether the data is better as a categorical (eg, by counting unique values in the column). The schema (at this level) is simply a hint that indicates that the original H5AD used a categorical, and that is of mixed value at best (ie, it is semi-random whether or not it is most efficiently encoded as a categorical).

Reply:
This is a stopgap before remaking the CXGs such that categoricals are explicitly defined in the schema and saved as integer arrays with the code-to-value dictionary. I think the determination of whether an array is / should be categorical is best done during the CXG creation process. This way, we also avoid needing to run the costly `unique` operation each time an `obs` column is fetched.